src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordPrintMembers.cs

```diff
@@ -113,7 +119,7 @@ internal override void GenerateMethodBody(TypeCompilationState compilationState,
                     return;
                 }
 
-                ArrayBuilder<BoundStatement>? block = printableMembers.IsEmpty ? null : ArrayBuilder<BoundStatement>.GetInstance();
+                ArrayBuilder<BoundStatement>? block = (printableMembers.IsEmpty && !ContainingType.IsRecordStruct) ? null : ArrayBuilder<BoundStatement>.GetInstance();
```

Comment:
>&& !ContainingType.IsRecordStruct [](start = 80, length = 33)

Addition of this condition looks suspicious. Are we going to return from the added `if` below without releasing the builder?

Reply:
Thanks. That was indeed a problem.

---
In reply to: [601677092](https://github.com/dotnet/roslyn/pull/52012#discussion_r601677092) [](ancestors = 601677092)